packages/docs/src/components/DocSiteNav.tsx

```diff
@@ -1,9 +1,87 @@
 import React from 'react';
+import { VerticalNav } from '@cmsgov/design-system';
+import { VerticalNavItemProps } from '@cmsgov/design-system/src/components/VerticalNav/VerticalNavItem';
+import { useStaticQuery, graphql } from 'gatsby';
 
+interface NavItem {
+  id: string;
+  name: string;
+  relativeDirectory: string;
+  relativePath: string;
+}
+
+/**
+ * DocSiteNav
+ * queries all MDX files and generates the proper format to create a vertical nav
+ * @returns {React Element}
+ * @todo figure out which item is currently selected and mark & expand appropriately
+ * @todo Determine URL string for each item once content is integrated
+ */
 const DocSiteNav = () => {
+  const data = useStaticQuery(graphql`
+    query SiteNavQuery {
+      allFile(sort: { fields: [relativeDirectory, name] }) {
+        group(field: relativeDirectory) {
+          fieldValue
+          edges {
+            node {
+              id
+              name
+              relativeDirectory
+              relativePath
+            }
+          }
+        }
+      }
+    }
+  `);
+
+  /**
+   * Updating a name to remove kebab case & get rid of numeric ordering
+   */
+  const formatNavItemLabel = (name: string): string => {
+    let newName = name.replace(/-/g, ' ');
+    newName = newName.replace(/\d+_/g, '');
+    return newName;
+  };
+
+  const formatNavItemData = ({ name, id }: NavItem) => ({
+    label: formatNavItemLabel(name),
+    url: '',
```

Comment:
Will this part be filled in later?

Reply:
Yes, I'm not quite clear on how URLs will work on the site yet, so once we start adding content pages & content, we should be able to fill this in